lib/hobo_fields/model/field_spec.rb

```diff
@@ -2,15 +2,35 @@
 
 module HoboFields
   module Model
-
     class FieldSpec
+      class UnknownSqlTypeError < RuntimeError; end
 
-      UTF8_BYTES_PER_CHAR = 3
+      MYSQL_TINYTEXT_LIMIT    =        0xff
+      MYSQL_TEXT_LIMIT        =      0xffff
+      MYSQL_MEDIUMTEXT_LIMIT  =   0xff_ffff
+      MYSQL_LONGTEXT_LIMIT    = 0xffff_ffff
+
+      class << self
+        # method for easy stubbing in tests
+        def mysql_text_limits?
+          if defined?(@mysql_text_limits)
```

Comment:
Styling nit here, for memoizing values where I want to respect `nil` I prefer this pattern
```ruby
@mysql_text_limits ||= unless defined?(@mysql_text_limits)
                         ActiveRecord::Base.connection.class.name.match?(/mysql/i)
                       end
```

Reply:
Hmm. That does look pretty succinct, but it's a bit redundant. It flattens into this equivalent:
```
if @mysql_text_limits
  @mysql_text_limits
else
  unless defined?(@mysql_text_limits)
    @mysql_text_limits = ActiveRecord::Base.connection.class.name.match?(/mysql/i)
  end
end
```
So when it's `false`, it will always wind up doing that double test. 

If lines of code are critical, we could always use an early `return`: :)
```
return @mysql_text_limits if defined?(mysql_text_limits)
@mysql_text_limits = ActiveRecord::Base.connection.class.name.match?(/mysql/i)
```
but I'm inclined to leave it.